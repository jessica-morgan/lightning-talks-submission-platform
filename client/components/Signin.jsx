import React from 'react'
import { Redirect, Link, withRouter } from 'react-router-dom'
import { connect } from 'react-redux'
import { signin } from '../actions/auth'
import Grid from '@material-ui/core/Grid'
import TextField from '@material-ui/core/TextField'
import Button from '@material-ui/core/Button'

class Signin extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      email: '',
      password: ''
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({
      [name]: value
    })
  }

  handleSubmit = (e) => {
    const user = this.state
    this.props.dispatch(signin(user))
    e.preventDefault()
  }

  render () {
    if (this.props.loggedIn) {
      return <Redirect to='/home' />
    }

    const { email, password } = this.state
    return (
      <div>
        <form>
          <Grid container direction="column" justify="center" alignItems="center">
            <TextField
              label="email"
              margin="normal"
              name="email"
              variant="outlined"
              onChange={this.handleChange}></TextField>
            <TextField
              label="password"
              margin="normal"
              name="password"
              variant="outlined"
              type="password"
              onChange={this.handleChange} ></TextField>
            <Button variant="contained" color="primary" onClick={this.handleSubmit}>
              Login
            </Button>
            <h3>Don't have an account? </h3>
            <Link to='/register'><Button variant="contained" color="primary" >
              Register
            </Button>
            </Link>
          </Grid>
        </form>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    loggedIn: state.auth.loggedIn,
    error: state.auth.error
  }
}

export default withRouter(connect(mapStateToProps)(Signin))
