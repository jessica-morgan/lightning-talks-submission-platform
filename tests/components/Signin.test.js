import React from 'react'
import Signin from '../../client/components/Signin'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

test('Signin renders correctly', () => {
  const mockStore = configureStore([thunk])({
    auth: [{ loggedIn: true, error: false }]
  })

  const component = renderer.create(
    <Provider store={mockStore}>
      <MemoryRouter>
        <Signin />
      </MemoryRouter>
    </Provider>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
