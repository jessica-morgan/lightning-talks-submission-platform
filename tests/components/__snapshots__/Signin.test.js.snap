// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin renders correctly 1`] = `
<div>
  <form>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      style={
        Object {
          "background": "rgb(210, 210, 210)",
          "borderRadius": "0.5vw",
          "height": "70vh",
          "margin": "auto",
          "marginTop": "13vh",
          "opacity": "0.8",
          "width": "30vw",
        }
      }
    >
      <h3
        style={
          Object {
            "color": "black",
            "fontSize": "2.5vh",
            "fontWeight": "lighter",
          }
        }
      >
        Please sign in 
      </h3>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
        style={
          Object {
            "background": "white",
            "borderRadius": "0.5vw",
          }
        }
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
        >
          email
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            name="email"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-163 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-164"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
        style={
          Object {
            "background": "white",
            "borderRadius": "0.5vw",
          }
        }
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
        >
          password
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            name="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="password"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-163 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-164"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "color": "black",
            "marginBottom": "2vh",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Login
        </span>
      </button>
      <div />
      <h3
        style={
          Object {
            "color": "black",
            "fontSize": "2.5vh",
            "fontWeight": "lighter",
          }
        }
      >
        Don't have an account? 
      </h3>
      <a
        href="/register"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "color": "black",
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Register
          </span>
        </button>
      </a>
    </div>
  </form>
</div>
`;
